test/cluster/mzcompose.py

```diff
@@ -5140,12 +5140,12 @@ def workflow_test_constant_sink(c: Composition) -> None:
                 > SELECT write_frontier
                   FROM mz_internal.mz_frontiers
                   JOIN mz_sinks ON id = object_id
-                  WHERE name = 'snk'
+                  WHERE name = 'snk' AND write_frontier IS NOT NULL
 
                 > SELECT status
                   FROM mz_internal.mz_sink_statuses
                   WHERE name = 'snk'
-                dropped
```

Comment:
Previously, the read frontier was immediately set to {} for sinks, so having the write frontier progress to {} would mark the sink dropped. After this PR, the read frontier can evolve independently, so the sink is not immediately removed from introspection.

Looking at https://github.com/MaterializeInc/database-issues/issues/8842 - this seems like an improvement, though there may be more to do. (In the future, not on this PR.)